### DISCUSSION TITLE
How to install DuckDB on Windows 10 in R with multiprocessing enabled

### DISCUSSION THREAD

**Discussion by AdrianAntico** [duckdb/duckdb#12312] [General]
I've been working on a benchmark that included DuckDB and run times are really slow. I only installed DuckDB in R using the simple ```install.packages("duckdb")``` command. Is there a way to enable multiprocessing? Using the DBI package I have attempted to tell DuckDB to use all cores but that doesn't seem to be working.
```
con = dbConnect(duckdb::duckdb())
ncores = parallel::detectCores()
invisible(dbExecute(con, sprintf("PRAGMA THREADS=%d", ncores)))
invisible(dbExecute(con, sprintf("SET THREADS=%d", ncores)))
```

**Comment by eitsupi** [comment]
What is the output of `SELECT current_setting('threads');`?
https://duckdb.org/docs/sql/statements/set

**Comment by AdrianAntico** [comment]
I get an output of 0

```
library(duckdb)
library(DBI)

con = dbConnect(duckdb::duckdb())
ncores = parallel::detectCores()
MaxMem = '243GB'
invisible(dbExecute(con, sprintf("PRAGMA THREADS=%d", ncores)))
invisible(dbExecute(con, sprintf("SET THREADS=%d", ncores)))
invisible(dbExecute(con, sprintf("SET memory_limit=%s", paste0("'", MaxMem, "'"))))
dbExecute(con, "SELECT current_setting('threads')")
```

**Reply by eitsupi** [reply]
Please run `duckdb:::sql("SELECT current_setting('threads')")`

**Comment by AdrianAntico** [comment]
Here you go! The main reason I'm asking about this is because I'm getting some slow runtimes on a benchmark that includes DuckDB...

https://github.com/AdrianAntico/Benchmarks

```
invisible(dbExecute(con, sprintf("PRAGMA THREADS=%d", ncores)))
invisible(dbExecute(con, sprintf("SET THREADS=%d", ncores)))
invisible(dbExecute(con, sprintf("SET memory_limit=%s", paste0("'", MaxMem, "'"))))
duckdb:::sql("SELECT current_setting('threads')")
#  current_setting('threads')
# 1                         64
```

**Reply by eitsupi** [reply]
Why not try other builds like CLI and WSL2 to see if it's really an R issue or a Windows specific issue?
